src/main/java/com/mparticle/kits/SkyhookKit.java

```diff
@@ -0,0 +1,170 @@
+package com.mparticle.kits;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+
+import com.mparticle.internal.MPUtility;
+import com.skyhookwireless.accelerator.AcceleratorClient;
+import com.skyhookwireless.accelerator.AcceleratorStatusCodes;
+
+import java.util.List;
+import java.util.Map;
+
+public class SkyhookKit
+    extends KitIntegration
+    implements KitIntegration.ActivityListener,
+               AcceleratorClient.ConnectionCallbacks,
+               AcceleratorClient.OnConnectionFailedListener,
+               AcceleratorClient.OnRegisterForCampaignMonitoringResultListener,
+               AcceleratorClient.OnStartCampaignMonitoringResultListener,
+               AcceleratorClient.OnStopCampaignMonitoringResultListener {
+
+    private static final String API_KEY = "apiKey";
+
+    private AcceleratorClient _client;
+    private boolean _isInitialized;
+
+    @Override
+    public String getName() {
+        return "Skyhook";
+    }
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(final Map<String, String> settings,
+                                                 final Context context) {
+        final String apiKey = settings.get(API_KEY);
+        new SkyhookPreferences(context).setApiKey(apiKey);
+
+        _client = new AcceleratorClient(context, apiKey, this, this);
+        SkyhookLog.i("Accelerator SDK v" + _client.getVersion());
+        initialize();
+        return null;
+    }
+
+    @Override
+    protected void onKitDestroy() {
+        _client.stopMonitoringForAllCampaigns(this);
```

Comment:
I noticed that the bootreceiver checks if API key is null to determine if it should be used - should you be nulling out the API key here?